deployment/clouddeploy/deployer.go

```diff
@@ -1694,10 +1701,40 @@ func (p *Deployer) Cleanup(ctx context.Context) error {
 				continue
 			}
 
-			p.removeCluster(ctx, cluster)
+			err := p.removeCluster(ctx, cluster)
+			if err != nil {
+				allErr = multierr.Append(allErr, errors.Wrapf(err, "cluster_id: %s", cluster.Cluster.Id))
+			}
 		}
 	}
 
+	projects, err := p.client.ListProjects(ctx, p.tenantID, &capellacontrol.PaginatedRequest{
+		Page:          1,
+		PerPage:       100,
+		SortBy:        "name",
+		SortDirection: "asc",
+	})
+
+	if err != nil {
+		allErr = multierr.Append(allErr, errors.Wrap(err, "failed to list all projects"))
+	}
+
+	for _, project := range projects.Data {
+		if !strings.HasPrefix(project.Data.Name, "cbdc2_") && project.Data.ClusterCount == 0 {
```

Comment:
This logic is wrong, it only delete projects that aren't ours, as ive unfortunately just discovered...